chia/simulator/add_blocks_in_batches.py

```diff
@@ -17,21 +17,28 @@
 async def add_blocks_in_batches(
     blocks: list[FullBlock],
     full_node: FullNode,
+    # if we're adding the block to a fork, this is the fork point
     header_hash: Optional[bytes32] = None,
 ) -> None:
-    if header_hash is None:
+    peak_hash = blocks[0].prev_header_hash
+    if blocks[0].height == 0:
+        assert peak_hash == full_node.constants.GENESIS_CHALLENGE
         diff = full_node.constants.DIFFICULTY_STARTING
         ssi = full_node.constants.SUB_SLOT_ITERS_STARTING
+        assert header_hash is None or header_hash == peak_hash
         fork_height = -1
-        fork_info = ForkInfo(-1, fork_height, full_node.constants.GENESIS_CHALLENGE)
     else:
+        # if no fork point is specified, assume it's immediately before the
+        # batch of block we're about to add
+        if header_hash is None:
```

Comment:
this means we can remove the header_hash param and just always take blocks[0].prev_header_hash if blocks[0].height == 0